### PR TITLE
Autoid collecting

### PR DESCRIPTION
**Pull request by weanti**
Added autoid collecting tool that gathers potential autoids from source files and from content files
Adapted autoid initialization

### CODE REVIEW

File: tools/aicollector/main.cpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+// append autoids to the given vector
+// - recursively parses a json "tree" and stores all values of string type objects
+// - strings with leading '%' are stored without '%' as well
+void parseJSON(Json::Value& root, std::set<std::string>& autoids)
+{
+    for ( auto it = root.begin(), e = root.end(); it != e; ++it)
```
Comment (HalalUr) [inline tools/aicollector/main.cpp]: for ( auto const& val : root ) or auto& val:root works with json::value too (possibly auto&& is good too).

File: run/CMakeLists.txt [hunk 2]
```diff
@@ -13,35 +13,73 @@ if(CMAKE_CROSSCOMPILING)
...
+        COMMAND ${SETENV_FOR_TOOLS}
+                ${AICOLLECTOR_VAR}
+                    --path ${CMAKE_CURRENT_SOURCE_DIR}/data
+                    --ext json
+                    --enable-json
+                    --output-file ${CMAKE_CURRENT_SOURCE_DIR}/data/autoids.content
+        COMMAND ${SETENV_FOR_TOOLS}
+                ${AICOLLECTOR_VAR}
+                    --path ${CMAKE_CURRENT_SOURCE_DIR}/../src
+                    --ext cpp h
```
Comment (HalalUr) [inline run/CMakeLists.txt:64]: You wrote comma separated. Maybe I just don't know what's happening

**Review by HalalUr — APPROVED** [review]
I can't fully review this PR because of the CMakeLists part. We need a CMAKE expert here

File: run/CMakeLists.txt [hunk 3]
```diff
@@ -13,35 +13,73 @@ if(CMAKE_CROSSCOMPILING)
...
+                ${PROJECT_SOURCE_DIR}/build
+                ${PROJECT_SOURCE_DIR}/local
+                ${REAPING2_DEPS_INSTALL_DIR}
+            PATH_SUFFIXES
+                bin
+                tools/aicollector)
+    if (NOT AICOLLECTOR_VAR)
+        message(STATUS "aicollector not found! Autoids won't be collected.")
+    else (NOT AICOLLECTOR_VAR)
+        message(STATUS "aicollector found! Autids will be collected.")
```
Comment (MrPepperoni) [inline run/CMakeLists.txt]: O

File: run/CMakeLists.txt (hunk 2 above)
Comment (MrPepperoni) [reply run/CMakeLists.txt:64]: the help text is incorrect, the usage is fine

File: run/CMakeLists.txt [hunk 4]
```diff
@@ -13,35 +13,73 @@ if(CMAKE_CROSSCOMPILING)
...
+if(R2PKG_VAR)
+    #data.pkg generating
     add_custom_command(OUTPUT data.pkg
         COMMAND
             ${SETENV_FOR_UPDATE_DATA}
             ${R2PKG_VAR}
                 -c ${CMAKE_CURRENT_SOURCE_DIR}/data
                 -o ${PROJECT_BINARY_DIR}/bin/$<LOWER_CASE:$<CONFIG>>/data.pkg
-        DEPENDS ${data_src})
+        DEPENDS ${content})
```
Comment (MrPepperoni) [inline run/CMakeLists.txt]: data.pkg depends on autoids, otherwise packaging might happen before generation

**Review by MrPepperoni — CHANGES_REQUESTED** [review]
dependencies need fixing

File: run/CMakeLists.txt (hunk 2 above)
Comment (weanti) [reply run/CMakeLists.txt:64]: The description is wrong, I'll update the help text

File: run/CMakeLists.txt (hunk 3 above)
Comment (weanti) [reply run/CMakeLists.txt]: Maybe a bit more specific? Use Error instead of Status?

Comment (MrPepperoni) [reply run/CMakeLists.txt]: Aut_O_ids will be collected.

Comment (weanti) [reply run/CMakeLists.txt]: Ah, okay. Thanks

**Review by MrPepperoni — APPROVED** [review]

**Pull request merged**